lib/core/shell/step.sh

```diff
@@ -149,57 +149,108 @@ If you want to source IH scripts earlier, adjust your .zshrc"
   fi
 }
 
-function ih::setup::core.shell::private::configure-profile() {
-
-  re_source
-
-  if ih::setup::core.shell::private::validate-profile; then
-    return 0
+function ih::setup::core.shell::private::collect-env-var() {
+  local var_name="$1"
+  local prompt_msg="$2"
+  local default_val="$3"
+  local input_val=""
+
+  # If the variable is already set, skip
+  if [[ -z ${!var_name} ]]; then
+    while [[ -z $input_val && -z $default_val ]]; do
+      read -p "$prompt_msg [$default_val]: " input_val
+      # If the input is empty and there's no default, keep looping
+      if [[ -z $input_val && -z $default_val ]]; then
+        echo "This value cannot be left empty. Please provide a value."
+      fi
+    done
+    # Use the default value if the input is empty
+    export $var_name="${input_val:-$default_val}"
   fi
+}
+
+function ih::setup::core.shell::private::configure-profile() {
+  echo "Please enter the requested information for each prompt."
+
+  ih::setup::core.shell::private::collect-env-var "EDITOR" \
+    "Your EDITOR is unset. What editor do you like to use? (maybe enter vim or nano, or 'code -w' to use VSCode)" \
+    ""
+
+  ih::setup::core.shell::private::collect-env-var "IH_HOME" \
+    "Directory where you want to clone Legacy Grand Rounds repos" \
+    "$HOME/src/github.com/ConsultingMD"
+  export GR_HOME="$IH_HOME"
+  ih::setup::core.shell::private::collect-env-var "EMAIL_ADDRESS" \
+    "Your Included Health email address" \
+    ""
+
+  ih::setup::core.shell::private::collect-env-var "GITHUB_USER" \
+    "Your GitHub username" \
+    ""
+  local default_email="$EMAIL_ADDRESS"
+  ih::setup::core.shell::private::collect-env-var "GITHUB_EMAIL_ADDRESS" \
+    "The email address you want to associate with commits. If you want to keep \
+    your email address private, or have configured your email address to be \
+    protected in GitHub, follow the guidance at \
+    https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address \
+    and put the no-reply email address here. Otherwise, you can leave this as is." \
+    "$default_email"
+
+  ih::setup::core.shell::private::collect-env-var "FULL_NAME" \
+    "Your full name, the name you would introduce yourself with" \
+    ""
+  ih::setup::core.shell::private::collect-env-var "IH_USERNAME" \
+    "Your username, probably firstname.lastname" \
+    ""
+  export GR_USERNAME="$IH_USERNAME"
+  local default_jira_username="$GR_USERNAME@includedhealth.com"
+  ih::setup::core.shell::private::collect-env-var "JIRA_USERNAME" \
+    "The username you have in JIRA. Before 1/15/2022: If you've already logged in \
+    to JIRA with an email address, use that. Otherwise, if you're new, use \
+    \"$GR_USERNAME@grandrounds.com\". After 7/17/2022: Use \
```

Comment:
Was thinking maybe it would still be relevant to old employees setting up new laptops like @cobbr2 and @fred-vogt-dod recently had to do.

But my jira username is actually using the new `@includedhealth.com` domain, so I guess it's not relevant anymore.

Reply:
Fixed: https://github.com/ConsultingMD/homebrew-ih-public/pull/65/commits/a6fb9fa28aba45c7040f268efc7731002afedaa4